packages/koenig-lexical/src/nodes/ToggleNodeComponent.jsx

```diff
@@ -0,0 +1,45 @@
+import CardContext from '../context/CardContext';
+import React, {useState} from 'react';
+import {$getNodeByKey} from 'lexical';
+import {ToggleCard} from '../components/ui/cards/ToggleCard';
+import {useLexicalComposerContext} from '@lexical/react/LexicalComposerContext';
+
+export function ToggleNodeComponent({nodeKey, content, contentPlaceholder, header, headerPlaceholder}) {
+    const [editor] = useLexicalComposerContext();
+    const cardContext = React.useContext(CardContext);
+
+    const [isContentVisible, setContentVisible] = useState(false);
+
+    const toggleContent = () => {
+        setContentVisible(!isContentVisible);
+    };
+
+    const setHeader = (newHeader) => {
+        editor.update(() => {
+            const node = $getNodeByKey(nodeKey);
+            node.setHeader(newHeader);
+        });
+    };
+
+    const setContent = (newContent) => {
+        editor.update(() => {
+            const node = $getNodeByKey(nodeKey);
+            node.setContent(newContent);
+        });
+    };
+
+    return (
+        <ToggleCard
```

Comment:
I'm not sure; maybe it's too early for such a comment as you're still working on the card, but just in case :)

It would be nice to add `ActionToolbar` here as in the current editor 
<img width="771" alt="Screenshot 2023-03-23 at 12 02 16" src="https://user-images.githubusercontent.com/14221798/227140066-17685d01-559e-46e3-8425-4dabd8478a35.png">

Example in audio card https://github.com/TryGhost/Koenig/blob/main/packages/koenig-lexical/src/nodes/AudioNode.jsx#L129-L138